Source/ExecutionEngine/ExecutionEngine.cs

```diff
@@ -243,41 +236,42 @@ public ExecutionEngine(ExecutionEngineOptions options, VerificationResultCache c
     static readonly ConcurrentDictionary<string, CancellationTokenSource> RequestIdToCancellationTokenSource =
       new ConcurrentDictionary<string, CancellationTokenSource>();
 
-    static ThreadTaskScheduler Scheduler = new ThreadTaskScheduler(16 * 1024 * 1024);
-
-    static TextWriter ModelWriter = null;
+    static ThreadTaskScheduler LargeStackScheduler = new ThreadTaskScheduler(16 * 1024 * 1024);
 
-    public bool ProcessFiles(IList<string> fileNames, bool lookForSnapshots = true, string programId = null)
-    {
+    public bool ProcessFiles(TextWriter output, IList<string> fileNames, bool lookForSnapshots = true,
```

Comment:
Why not make this function async as well?

Reply:
I didn't need that. It's not called from Dafny, only from BoogieDriver